src/sql-parser/src/parser.rs

```diff
@@ -4540,13 +4556,47 @@ impl<'a> Parser<'a> {
     }
 
     fn parse_set_variable_value(&mut self) -> Result<SetVariableValue, ParserError> {
-        let token = self.peek_token();
-        Ok(match (self.parse_value(), token) {
-            (Ok(value), _) => SetVariableValue::Literal(value),
-            (Err(_), Some(Token::Keyword(DEFAULT))) => SetVariableValue::Default,
-            (Err(_), Some(Token::Keyword(kw))) => SetVariableValue::Ident(kw.into_ident()),
```

Comment:
@jkosh44 / @benesch any idea why this is needed? For example
```sql
set search_path to select;
```
is accepted by Materialize, but is rejected by Postgres, so it's not an issue fo Postgres-compatibility.

Reply:
OK, I think this might have something to do with it:
```sql
SET standard_conforming_strings = ON;
```

Reply:
Postgres and us allow the value to not have quotes around it in certain cases. For example we both allow

```SQL
SET TRANSACTION_ISOLATION TO serializable;
```
Without that line of code we'd reject all values without quotes. Clearly though we're a bit too lenient and allow too many unquoted values compared to Postgres. I can open an issue for this. Is it a blocker for this PR?

Reply:
> SET standard_conforming_strings = typo;

This will be covered by the `Token::Ident` case. I think the problem is that some unquoted strings (such as `ON`) will be tokenized as a `Token::Keyword` instead of a `Token::Identifier`, which is why you need the `SetVariableValue::Ident(kw.into_ident())` case.